src/core/manager.c

```diff
@@ -1973,18 +1974,28 @@ int manager_load_unit_prepare(
 
         ret = manager_get_unit(m, name);
         if (ret) {
-                *_ret = ret;
-                return 1;
+                /* The time-based cache allows to start new units without daemon-reload,
+                 * but if they are already referenced (because of dependencies or ordering)
+                 * then we have to force a load of the fragment. As an optimization, check
+                 * first if anything in the usual paths was modified since the last time
+                 * the cache was loaded. */
+                if (ret->load_state == UNIT_NOT_FOUND && m->unit_cache_mtime > 0 &&
+                                !lookup_paths_mtime_good(&m->lookup_paths, m->unit_cache_mtime)) {
```

Comment:
hmm, the mtime check should be encapsulated in a helper call of its own I am sure:

```
int manager_unit_cache_needs_refresh(Manager *m);
```

or so.